client/state/data-layer/wpcom/sites/automated-transfer/eligibility/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	AUTOMATED_TRANSFER_ELIGIBILITY_REQUEST,
+} from 'state/action-types';
+
+import { updateEligibility } from 'state/automated-transfer/actions';
+import {
+	AUTOMATED_TRANSFER_STATUS,
+	eligibilityHolds,
+} from 'state/automated-transfer/constants';
+
+const eligibilityUrl = domain =>
+	`https://public-api.wordpress.com/rest/v1/sites/${ domain }/automated-transfers/eligibility`;
+
+const statusMapping = {
+	no_wpcom_nameservers: eligibilityHolds.NO_WPCOM_NAMESERVERS,
+	not_using_custom_domain: eligibilityHolds.NOT_USING_CUSTOM_DOMAIN,
+	non_admin_user: eligibilityHolds.NON_ADMIN_USER,
+	site_graylisted: eligibilityHolds.SITE_GRAYLISTED,
+	site_private: eligibilityHolds.SITE_PRIVATE,
+};
+
+const fromApi = data => ( {
+	lastUpdate: Date.now(),
+	status: AUTOMATED_TRANSFER_STATUS.INQUIRING,
+	eligibilityHolds: get( statusMapping, data.status_code, [] ),
+	eligibilityWarnings: [
+		...get( data, 'warnings.plugins', [] ),
+		...get( data, 'warnings.widgets', [] ),
+	].map( ( { description, name, support_url } ) => ( { name, description, supportUrl: support_url } ) ),
+} );
+
+const receiveEligibility = ( dispatch, domain ) => data =>
+	dispatch( updateEligibility( domain, fromApi( data ) ) );
+
+export const fetchEligibility = ( { dispatch }, { domain } ) =>
+	fetch( eligibilityUrl( domain ) )
```

Comment:
thanks @aduth. this I knew and I don't plan on putting this out there to boot. mainly I wanted to see how sending a fetch directly would work with authentication and such. I would love to see this turn into a declarative HTTP request

```js
dispatch( {
	type: HTTP_REQUEST,
	method: HTTP_GET,
	url: eligibilityUrl( domain ),
	onSuccess: AUTOMATED_TRANSFER_ELIGIBILITY_REQUEST,
	onError: AUTOMATED_TRANSFER_ELIGIBILITY_REQUEST,
} );
```

and the HTTP middleware would run it, then dispatch those associated return actions with the following structure:

```js
type: AUTOMATED_TRANSFER_ELIGIBILITY_REQUEST,
domain,
meta: {
	httpResponse: {
		type: HTTP_SUCCESS | HTTP_ERROR,
		data: [ data returned from request if success ],
		error: [ error returned from request if error ],
	}
}
```

Note that this includes the original action data and the HTTP meta which should allow us to reuse the action type responsibly and also couple the request and the response.